exercises/grade-school/canonical-data.json

```diff
@@ -1,113 +1,241 @@
+      "input": {
+        "students": [["Blair", 2], ["James", 2], ["James", 2], ["Paul", 2]]
+      },
+      "expected": [true, true, false, true]
+    },
+    {
+      "uuid": "c125dab7-2a53-492f-a99a-56ad511940d8",
+      "description": "A student can't be in two different grades",
+      "property": "roster",
+      "input": {
+        "students": [["Aimee", 2], ["Aimee", 1]],
+        "desiredGrade": 2
+      },
+      "expected": []
+    },
+    {
+      "uuid": "a0c7b9b8-0e89-47f8-8b4a-c50f885e79d1",
+      "reimplements": "c125dab7-2a53-492f-a99a-56ad511940d8",
+      "comments": [
+        "Reimplemented to be logically consistent"
+      ],
+      "description": "A student can only be added to the same grade in the roster once",
+      "property": "roster",
+      "input": {
+        "students": [["Aimee", 2], ["Aimee", 2]]
+      },
+      "expected": ["Aimee"]
+    },
+    {
+      "uuid": "d7982c4f-1602-49f6-a651-620f2614243a",
+      "description": "Student not added to same grade in the roster more than once",
+      "reimplements": "a0c7b9b8-0e89-47f8-8b4a-c50f885e79d1",
+      "comments": [
+        "Ensure the implementation does not affect students already added",
+        "Ensure the implementation does not prevent addition of other students after preventing the error case"
+      ],
+      "property": "roster",
+      "input": {
+        "students": [["Blair", 2], ["James", 2], ["James", 2], ["Paul", 2]]
+      },
+      "expected": ["Blair", "James", "Paul"]
+    },
+    {
+      "uuid": "e70d5d8f-43a9-41fd-94a4-1ea0fa338056",
+      "description": "Adding students in multiple grades",
+      "property": "add",
+      "input": {
+        "students": [["Chelsea", 3], ["Logan", 7]]
+      },
+      "expected": [true, true]
+    },
+    {
+      "uuid": "75a51579-d1d7-407c-a2f8-2166e984e8ab",
+      "description": "Students in multiple grades are added to the roster",
+      "property": "roster",
+      "input": {
+        "students": [["Chelsea", 3], ["Logan", 7]]
+      },
+      "expected": ["Chelsea", "Logan"]
+    },
+    {
+      "uuid": "7df542f1-57ce-433c-b249-ff77028ec479",
+      "description": "Cannot add same student to multiple grades in the roster",
+      "property": "add",
+      "input": {
+        "students": [["Blair", 2], ["James", 2], ["James", 3], ["Paul", 3]]
+      },
+      "expected": [true, true, false, true]
+    },
+    {
+      "uuid": "6a03b61e-1211-4783-a3cc-fc7f773fba3f",
+      "reimplements": "c125dab7-2a53-492f-a99a-56ad511940d8",
+      "comments": [
+        "Reimplemented to be logically consistent"
+      ],
+      "description": "A student cannot be added to more than one grade in the sorted roster",
+      "property": "roster",
+      "input": {
+        "students": [["Aimee", 2], ["Aimee", 1]]
+      },
+      "expected": ["Aimee"]
+    },
+    {
+      "description": "Student not added to multiple grades in the roster",
+      "uuid": "c7ec1c5e-9ab7-4d3b-be5c-29f2f7a237c5",
```

Comment:
this case appears to uniquely be the one where `uuid` isn't first. obviously this isn't a problem as the keys can come in any order, but it is unusual.